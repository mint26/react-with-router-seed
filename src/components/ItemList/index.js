import React from 'react'; 

const itemList = (props) => {
    let items = props.items && props.items.length > 0 ? props.items.map((item, key) =>{
        return (
            <li className={props.itemCss} key={`item-${key}`}>{item.content}</li>
        );
    }) : null;
    return (
        <ul className={props.listCss}>
            {items}
        </ul>
    );
};

export default itemList; 