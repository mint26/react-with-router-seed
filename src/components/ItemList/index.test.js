import React from 'react';
import ItemList from './';
import renderer from 'react-test-renderer';

test('Link changes the class when hovered', () => { 
    const dummyItems = [
        {
            content : "<span>hello</span>"
        },
        {
            content : "<span>hello 2</span>"
        },
        {
            content : "<span>hello 3</span>"
        }
    ];
    const component = renderer.create(
    <ItemList items={dummyItems} itemCss="test-item" listCss="test-list"/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});