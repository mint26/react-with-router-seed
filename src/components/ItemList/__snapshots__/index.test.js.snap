// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<ul
  className="test-list"
>
  <li
    className="test-item"
  >
    &lt;span&gt;hello&lt;/span&gt;
  </li>
  <li
    className="test-item"
  >
    &lt;span&gt;hello 2&lt;/span&gt;
  </li>
  <li
    className="test-item"
  >
    &lt;span&gt;hello 3&lt;/span&gt;
  </li>
</ul>
`;
